javaparser-core/src/main/java/com/github/javaparser/ast/Node.java

```diff
@@ -677,4 +678,50 @@ protected SymbolResolver getSymbolResolver() {
     // We need to expose it because we will need to use it to inject the SymbolSolver
     public static final DataKey<SymbolResolver> SYMBOL_RESOLVER_KEY = new DataKey<SymbolResolver>() {
     };
+
+    /**
+     * Walks the AST, calling the consumer for every node.
+     */
+    public void walk(Consumer<Node> consumer) {
+        new TreeVisitor() {
+            @Override
+            public void process(Node node) {
+                consumer.accept(node);
+            }
+        }.visitPreOrder(this);
```

Comment:
should be the order (preOrder/postOrder) a flag?

Reply:
Maybe, I didn't do that right now because it would make code less concise, assuming that pre-order is the most common case. You can always use the `TreeVisitor` for others. But I could make another overloaded method :-)